core/modules/layout/layout.theme.inc

```diff
@@ -236,9 +241,32 @@ function layout_preprocess_layout(&$variables) {
   }
 
   if (!$variables['admin']) {
-    $variables['action_links'] = menu_local_actions();
-    $variables['tabs'] =  menu_local_tabs();
+    if ($should_display['action_links']) {
+      $variables['action_links'] = menu_local_actions();
+    }
+    if ($should_display['tabs']) {
+      $variables['tabs'] =  menu_local_tabs();
+    }
+  }
+}
+
+/**
+ * Utility function to return if a title component block is in a layout.
+ */
+function _layout_page_component_blocks_exist($variables) {
+  $active_component_blocks = array();
+  foreach ($variables['layout']->content as $block) {
+    if ($block->delta == 'page_components') {
+      $active_component_blocks[] = $block->childDelta;
+    }
   }
+
+  $should_display['title'] = (!in_array('title_combo', $active_component_blocks) && !in_array('title', $active_component_blocks));
+  $should_display['messages'] = (!in_array('title_combo', $active_component_blocks) && !in_array('messages', $active_component_blocks));
+  $should_display['tabs'] = (!in_array('title_combo', $active_component_blocks) && !in_array('tabs', $active_component_blocks));
+  $should_display['action_links'] = (!in_array('title_combo', $active_component_blocks) && !in_array('action_links', $active_component_blocks));
+
+  return $should_display;
```

Comment:
The name of this function and it's return values are at odds with each other. The function name contains `_block_exists`, but what is returned is opposite that. The return value for `$should_display['title']` is `FALSE` if the block exists. That's confusing.

I think the function name is fine, but this should return an array of `$block_exists['title']` being TRUE if the block exists, then reverse all the conditions in `layout_preprocess_layout()`.

Either that or you can simply rename the function, e.g. `_layout_page_components_to_display()`